apps/server/src/shared/infra/identity-management/identity-management.interface.ts

```diff
@@ -0,0 +1,52 @@
+import { IAccount, IAccountUpdate } from '@shared/domain';
+
+export interface IIdentityManagement {
+	/**
+	 * Create a new account in the identity management.
+	 *
+	 * @param account the account's details
+	 * @param [password] the account's password (optional)
+	 * @returns the account id if created successfully
+	 */
+	createAccount(account: IAccount, password?: string | undefined): Promise<string>;
+
+	/**
+	 * Update an existing account's details.
+	 *
+	 * @param accountId the account to be updated.
+	 * @param account the account data to be applied.
+	 * @returns the account id if updated successfully
+	 */
+	updateAccount(accountId: string, account: IAccountUpdate): Promise<string>;
+
+	/**
+	 * Update an existing account's password.
+	 *
+	 * @param accountId the account to be updated.
+	 * @param password the new password (clear).
+	 * @returns the account id if updated successfully
+	 */
+	updateAccountPassword(accountId: string, password: string): Promise<string>;
```

Comment:
I don't if it's necessary, but we have a kind of value object type `EntityId` for entity identifiers, which wraps the raw string type.